pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixServerStarter.java

```diff
@@ -261,6 +306,22 @@ private void updateInstanceConfigIfNeeded(String host, int port) {
         "Failed to update instance config");
   }
 
+  // Update instance configs with environment specific properties
+  private void updateInstanceConfigWithEnvironmentSpecificConfigs() {
```

Comment:
Didn't we conclude in the design document that the instance configs will be set before HelixServerStarter is called?

Reply:
After proper consideration it seemed best to update instance configs at Server Starter layer. We are already doing something similar. Refer to: updateInstanceConfigIfNeeded method in HelixServerStarter.java. Updating configs before the call to HelixServerStarter is not possible as we initialize the EnvironmentProvider factory at HelixServerStarter layer. If we move the entire logic before the call to HelixServerStarter is made then we will be adding a lot of redundant code of connecting to helixManager and retrieving the helixAdmin instance. Hence, this solution.

Reply:
Updating the document for consistency.